### PR TITLE
t.sentinel.import: adapt for USGS download and sen2cor

### PR DESCRIPTION
**Pull request by griembauer**
This PR adapts `t.sentinel.import` as follows:

- extends t.sentinel.import to more free download options, also without `s2names`, but with `start`/`end`/`producttype`/`cloud` filters instead
- also allows for download from USGS (by `datasource` option)
- optionally runs atmospherical correction using `sen2cor` and the Grass Addon `i.sentinel-2.sen2cor` (especially useful when downloading from USGS, where only L1C data are available) 
- [added 08.03.2021]: Cloud and Shadow rasters from the same date+time but from different scenes (e.g. different UTM tiles) are patched together to a single raster respectively. This requires also adaptions to `t.rast.mosaic`: https://github.com/mundialis/t.rast.mosaic/pull/3

### CODE REVIEW

File: t.sentinel.import/t.sentinel.import.html [hunk 1]
```diff
@@ -15,11 +15,18 @@ <h2>SEE ALSO</h2>
...
 <a href="i.sentinel.import.worker.html">i.sentinel.import.worker</a>,
 <a href="t.create.html">t.create</a>,
 <a href="t.register.html">t.register</a>
 </em>
 
+<h2>REQUIREMENTS</h2>
+If atmospheric correction should be run using the <b>-a</b> flag,
+<a href="http://step.esa.int/main/snap-supported-plugins/sen2cor/">Sen2Cor</a>
+needs to be installed. For additional information, see
+<a href="i.sentinel-2.sen2cor.html">i.sentinel-2.sen2cor.html</a>.
```
Comment (neteler) [inline t.sentinel.import/t.sentinel.import.html:28]:
```suggestion
<a href="i.sentinel-2.sen2cor.html">i.sentinel-2.sen2cor</a>.
```

File: t.sentinel.import/t.sentinel.import.py [hunk 2]
```diff
@@ -43,19 +43,88 @@
...
 #%end
 
+#%flag
+#% key: e
+#% description: Use ESA-style scenename/s to download from USGS
+#%end
+
+#%flag
+#% key: a
+#% description: run atmospheric correction with sen2cor before importing
```
Comment (neteler) [inline t.sentinel.import/t.sentinel.import.py:53]:
```suggestion
#% description: Run atmospheric correction with sen2cor before importing
```

File: t.sentinel.import/t.sentinel.import.py [hunk 3]
```diff
@@ -43,19 +43,88 @@
...
+#% description: Maximum cloud cover percentage for Sentinel scene
+#% required: no
+#% guisection: Filter
+#% answer: 100
+#%end
+
+#%option
+#% key: producttype
+#% type: string
+#% description: Sentinel product type to filter
```
Comment (neteler) [inline t.sentinel.import/t.sentinel.import.py:74]:
```suggestion
#% description: Sentinel-2 product type to filter
```

... unless also S1 will be supported (?)

File: t.sentinel.import/t.sentinel.import.py [hunk 4]
```diff
@@ -43,19 +43,88 @@
...
+#% required: no
 #% multiple: yes
 #% description: List of Sentinel-2 names or file with this list
 #%end
 
+#%option
+#% key: sen2cor_path
+#% required: no
+#% type: string
+#% label: path to sen2cor home directory
```
Comment (neteler) [inline t.sentinel.import/t.sentinel.import.py:124]:
```suggestion
#% label: Path to sen2cor home directory
```

**Review by anikaweinmann — APPROVED** [review]

File: t.sentinel.import/t.sentinel.import.py [hunk 5]
```diff
@@ -43,19 +43,88 @@
...
 #%option G_OPT_F_INPUT
 #% key: settings
+#% required: yes
 #% label: Full path to settings file (user, password)
 #%end
 
+#%option
+#% key: clouds
+#% type: integer
+#% description: Maximum cloud cover percentage for Sentinel scene
```
Comment (neteler) [inline t.sentinel.import/t.sentinel.import.py:65]:
```suggestion
#% description: Maximum cloud cover percentage for Sentinel-2 scene
```

File: t.sentinel.import/t.sentinel.import.py [hunk 6]
```diff
@@ -43,19 +43,88 @@
...
+#%option
+#% key: end
+#% type: string
+#% description: End date ('YYYY-MM-DD')
+#% guisection: Filter
+#%end
+
+#%option
+#% key: datasource
+#% description: Data-Hub to download scenes from.
```
Comment (neteler) [inline t.sentinel.import/t.sentinel.import.py:97]:
```suggestion
#% description: Data-Hub to download scenes from
```

**Review by neteler — APPROVED** [review]

**Pull request merged**